Runtime error when try to enable secondaryVisibilityStore

The error starts after the job is finished. Operator doesn't start temporal cluster
```
2023-08-22T10:26:16Z	INFO	Starting reconciliation	{"controller": "temporalcluster", "controllerGroup": "temporal.io", "controllerKind": "TemporalCluster", "TemporalCluster": {"name":"temporal","namespace":"default"}, "namespace": "default", "name": "temporal", "reconcileID": "2bfe9792-2309-4599-b120-d8bb25f60ae0"}
2023-08-22T10:26:16Z	INFO	Observed a panic in reconciler: runtime error: invalid memory address or nil pointer dereference	{"controller": "temporalcluster", "controllerGroup": "temporal.io", "controllerKind": "TemporalCluster", "TemporalCluster": {"name":"temporal","namespace":"default"}, "namespace": "default", "name": "temporal", "reconcileID": "2bfe9792-2309-4599-b120-d8bb25f60ae0"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x203719c]

goroutine 330 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.1/pkg/internal/controller/controller.go:115 +0x1fa
panic({0x22732c0, 0x3eb1a90})
	/usr/local/go/src/runtime/panic.go:884 +0x213
github.com/alexandrevilain/temporal-operator/controllers.(*TemporalClusterReconciler).reconcilePersistence.func13({0x2b1efe0?, 0xc004080400?})
	/workspace/controllers/reconcile_persistence.go:165 +0x3c
github.com/alexandrevilain/controller-tools/pkg/reconciler.(*JosbReconciler).Reconcile(0xc000518810, {0x2b3dad8, 0xc004c2d1d0}, {0x2b59468?, 0xc004080400}, 0x2759fb8, {0xc004c4cba0, 0x9, 0x100?})
	/go/pkg/mod/github.com/alexandrevilain/controller-tools@v0.2.1/pkg/reconciler/jobs_reconcilier.go:55 +0x11f
github.com/alexandrevilain/temporal-operator/controllers.(*TemporalClusterReconciler).reconcilePersistence(0xc000534800, {0x2b3dad8, 0xc004c2d1d0}, 0xc004080400)
	/workspace/controllers/reconcile_persistence.go:251 +0x14dd
github.com/alexandrevilain/temporal-operator/controllers.(*TemporalClusterReconciler).Reconcile(0xc000534800, {0x2b3dad8, 0xc004c2d1d0}, {{{0xc0006a7098?, 0x0?}, {0xc0006a70d8?, 0x40e027?}}})
	/workspace/controllers/temporalcluster_controller.go:130 +0x594
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x2b3dad8?, {0x2b3dad8?, 0xc004c2d1d0?}, {{{0xc0006a7098?, 0x21ba8e0?}, {0xc0006a70d8?, 0x2b20638?}}})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.1/pkg/internal/controller/controller.go:118 +0xc8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000bca0a0, {0x2b3da30, 0xc000a31c70}, {0x2362f00?, 0xc0006ba860?})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.1/pkg/internal/controller/controller.go:314 +0x377
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000bca0a0, {0x2b3da30, 0xc000a31c70})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.1/pkg/internal/controller/controller.go:265 +0x1d9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.1/pkg/internal/controller/controller.go:226 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.1/pkg/internal/controller/controller.go:222 +0x587
```	


```
apiVersion: temporal.io/v1beta1
kind: TemporalCluster
metadata:
  name: temporal
spec:
  version: 1.21.4
  numHistoryShards: 3
  metrics:
    enabled: false
    prometheus:
      listenPort: 9090
      scrapeConfig:
        serviceMonitor:
          enabled: false
          labels: 
            release: monitoring
  dynamicConfig:
    pollInterval: 10s
    values:
      frontend.rps:
      - value: 4800
        constraints: {}
      frontend.namespaceCount:
      - value: 2400
        constraints: {}
      limit.blobSize.warn:
      - value: 2097152
        constraints: {}
      limit.blobSize.error:
      - value: 4194304
        constraints: {}
      frontend.workerVersioningDataAPIs:
      - value: true
        constraints: {}
      frontend.workerVersioningWorkflowAPIs:
      - value: true
        constraints: {}
      worker.buildIdScavengerEnabled:
      - value: true
        constraints: {}
  jobTtlSecondsAfterFinished: 300
  archival:
    enabled: true
    provider:
      s3:
        roleName: "iam_role"
        region: us-east-2
    history:
      enabled: true
      enableRead: true
      path: "bucket"
      paused: false
    visibility:
      enabled: true
      enableRead: true
      path: "bucket"
      paused: false
  persistence:
    defaultStore:
      sql:
        user: tmp
        pluginName: postgres
        databaseName: temporal
        connectAddr: rds.amazonaws.com:5432
        connectProtocol: tcp
      passwordSecretRef:
        name: temporal-service
        key: POSTGRES_PWD
    visibilityStore:
      sql:
        user: tmp
        pluginName: postgres
        databaseName: temporal_visibility
        connectAddr: rds.amazonaws.com:5432
        connectProtocol: tcp
      passwordSecretRef:
        name: temporal-service
        key: POSTGRES_PWD
    secondaryVisibilityStore:
      elasticsearch:
        version: v7
        username: tmp
        url: https://us-east-2.es.amazonaws.com
        enableHealthcheck: true
        indices:
          visibility: temporal_visibility
      passwordSecretRef:
        name: temporal-service
        key: OPENSEARCH_PWD
```

--- a/controllers/reconcile_persistence.go
+++ b/controllers/reconcile_persistence.go
@@ -49,6 +49,10 @@ func (r *TemporalClusterReconciler) reconcilePersistenceStatus(cluster *v1beta1.
 		cluster.Status.Persistence.VisibilityStore = new(v1beta1.DatastoreStatus)
 	}
 
+	if cluster.Status.Persistence.SecondaryVisibility == nil && cluster.Spec.Persistence.SecondaryVisibilityStore != nil {
+		cluster.Status.Persistence.SecondaryVisibility = new(v1beta1.DatastoreStatus)
+	}
+
 	if cluster.Status.Persistence.AdvancedVisibilityStore == nil && cluster.Spec.Persistence.AdvancedVisibilityStore != nil {
 		cluster.Status.Persistence.AdvancedVisibilityStore = new(v1beta1.DatastoreStatus)
 	}

